crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -20,41 +22,49 @@ pub(crate) fn parse<P: AsRef<Path>>(path_to_toml: P) -> BTreeMap<String, InputVa
 /// understands for Inputs
-fn toml_map_to_field(toml_map: BTreeMap<String, TomlTypes>) -> BTreeMap<String, InputValue> {
+fn toml_map_to_field(toml_map: BTreeMap<String, TomlTypes>) -> Result<BTreeMap<String, InputValue>, String> {
     let mut field_map = BTreeMap::new();
 
     for (parameter, value) in toml_map {
         match value {
             TomlTypes::String(string) => {
+                let new_value = parse_str(&string)?;       
                 let old_value =
-                    field_map.insert(parameter.clone(), InputValue::Field(parse_str(&string)));
-                assert!(old_value.is_none(), "duplicate variable name {}", parameter);
+                    field_map.insert(parameter.clone(), InputValue::Field(new_value));
+                if !old_value.is_none() {
+                    return Err(format!("duplicate variable name {}", parameter))
+                }
             }
             TomlTypes::Integer(integer) => {
+                let new_value = parse_str(&integer.to_string())?;
                 let old_value = field_map
-                    .insert(parameter.clone(), InputValue::Field(parse_str(&integer.to_string())));
-                assert!(old_value.is_none(), "duplicate variable name {}", parameter);
+                    .insert(parameter.clone(), InputValue::Field(new_value));
+                if !old_value.is_none() {
+                    return Err(format!("duplicate variable name {}", parameter))
+                }
             }
             TomlTypes::ArrayNum(arr_num) => {
-                let array_elements: Vec<_> =
+                let array_elements: Result<Vec<_>, _> =
                     arr_num.into_iter().map(|elem_num| parse_str(&elem_num.to_string())).collect();
-
                 let old_value =
-                    field_map.insert(parameter.clone(), InputValue::Vec(array_elements));
-                assert!(old_value.is_none(), "duplicate variable name {}", parameter);
+                    field_map.insert(parameter.clone(), InputValue::Vec(array_elements?));
+                if !old_value.is_none() {
+                    return Err(format!("duplicate variable name {}", parameter))
+                }           
             }
             TomlTypes::ArrayString(arr_str) => {
-                let array_elements: Vec<_> =
+                let array_elements: Result<Vec<_>, _> =
                     arr_str.into_iter().map(|elem_str| parse_str(&elem_str)).collect();
-
                 let old_value =
-                    field_map.insert(parameter.clone(), InputValue::Vec(array_elements));
-                assert!(old_value.is_none(), "duplicate variable name {}", parameter);
+                    field_map.insert(parameter.clone(), InputValue::Vec(array_elements?));
+                if !old_value.is_none() {
+                    return Err(format!("duplicate variable name {}", parameter))
+                }            
```

Comment:
Since this inserting and checking code is repeated in each branch it may help to break it out into another function. It could return a Result so you still get the early-return of errors: `my_helper(field_map, parameter, input_value)?;`